cdap-security-authz-dataset/src/main/java/co.cask.cdap.security.authz/ACLDataset.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.security.authz;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.AbstractDataset;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.data2.dataset2.lib.table.MDSKey;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.id.EntityId;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
```

Comment:
Would be good to add some note on how the table structure looks like. Row key byte array and columns. Looks like we give precedence for entities rather than users? So we can't perform an operation such as = give me entities that an user has access to?